packages/next/src/client/script.tsx

```diff
@@ -234,7 +234,7 @@ function Script(props: ScriptProps): JSX.Element | null {
 
       hasLoadScriptEffectCalled.current = true
     }
-  }, [props, strategy])
+  }, [props, (strategy === "afterInteractive"), strategy === "lazyOnload"])
```

Comment:
```
  useEffect(() => {
    if (!hasLoadScriptEffectCalled.current) {
      if (strategy === 'afterInteractive') {
        loadScript(props)
      } else if (strategy === 'lazyOnload') {
        loadLazyScript(props)
      }

      hasLoadScriptEffectCalled.current = true
    }
  }, [props, strategy])
```

Looking at the code it doesn't make sense to only run the effect on `strategy === 'afterInteractive'` or `strategy === 'lazyOnload'` because besides that `if` / `else if` condition there's a statement that always runs: `hasLoadScriptEffectCalled.current = true`. So you don't want it to only be affected by these two.